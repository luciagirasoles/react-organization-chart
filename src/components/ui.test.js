import React from "react";
import { render } from "@testing-library/react";

import { Button, Card, Input, Modal, Select } from "./ui";

test("Button", () => {
  expect(
    render(<Button>This is content</Button>).asFragment()
  ).toMatchSnapshot();
});

test("Card", () => {
  expect(render(<Card>This is content</Card>).asFragment()).toMatchSnapshot();
});

test("Input", () => {
  expect(render(<Input />).asFragment()).toMatchSnapshot();
});

test("Modal", () => {
  expect(render(<Modal />).asFragment()).toMatchSnapshot();
});

test("Select", () => {
  expect(
    render(
      <Select>
        <option value="testing">Testing</option>
      </Select>
    ).asFragment()
  ).toMatchSnapshot();
});
