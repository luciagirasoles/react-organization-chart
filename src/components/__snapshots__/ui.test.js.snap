// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `
<DocumentFragment>
  .emotion-0 {
  background-color: black;
  border: 1px solid black;
  border-radius: .25rem;
  color: white;
  cursor: pointer;
  font-size: .8rem;
  padding: .75rem 0;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
  text-align: center;
  text-transform: uppercase;
  width: 100%;
}

.emotion-0:hover {
  background-color: white;
  color: black;
}

<button
    class="emotion-0"
  >
    This is content
  </button>
</DocumentFragment>
`;

exports[`Card 1`] = `
<DocumentFragment>
  .emotion-0 {
  background: white;
  border-radius: .5em;
  box-shadow: 0px 2px 5px 0px rgba(0,0,0,.12);
  box-sizing: border-box;
  padding: 2rem;
  width: 100%;
}

<div
    class="emotion-0"
  >
    This is content
  </div>
</DocumentFragment>
`;

exports[`Input 1`] = `
<DocumentFragment>
  .emotion-0 {
  background: none;
  border: 1px solid #eaeaea;
  border-radius: .25rem;
  box-sizing: border-box;
  display: block;
  font-size: 1rem;
  padding: .5rem;
  width: 100%;
}

.emotion-0:focus {
  outline: none;
  border-color: #444444;
}

<input
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Modal 1`] = `
<DocumentFragment>
  .emotion-1 {
  background-color: rgba(0,0,0,.5);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 100;
}

.emotion-0 {
  position: fixed;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  width: 80%;
  max-width: 450px;
  z-index: 110;
}

<div
    class="emotion-1"
  >
    <div
      class="emotion-0"
      role="dialog"
    />
  </div>
</DocumentFragment>
`;

exports[`Select 1`] = `
<DocumentFragment>
  .emotion-2 {
  -webkit-appearance: none;
  color: rgb(0,0,0);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  height: 40px;
  font-size: 12px;
  text-transform: uppercase;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-weight: 100;
  position: relative;
  white-space: nowrap;
  line-height: 0;
  width: auto;
  min-width: 160px;
  background: rgb(255,255,255);
  outline: none;
  border-width: 1px;
  border-style: solid;
  border-color: rgb(234,234,234);
  border-image: initial;
  overflow: hidden;
  -webkit-transition: border 0.2s ease 0s,background 0.2s ease 0s,color 0.2s ease-out 0s,box-shadow 0.2s ease 0s;
  transition: border 0.2s ease 0s,background 0.2s ease 0s,color 0.2s ease-out 0s,box-shadow 0.2s ease 0s;
  border-radius: 4px;
}

.emotion-0 {
  height: 100%;
  box-shadow: none;
  color: rgb(0,0,0);
  line-height: 40;
  font-size: 14px;
  margin-right: -20px;
  width: calc(100% + 20px);
  text-transform: none;
  border-width: initial;
  border-style: none;
  border-color: initial;
  border-image: initial;
  background: none transparent;
  padding: 0px 76px 0px 16px;
  outline: none;
}

.emotion-1 {
  width: 30px;
  height: 100%;
  position: absolute;
  right: 0;
  pointer-events: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-left: 1px solid rgb(234,234,234);
  background: rgb(255,255,255);
  -webkit-transition: border 0.2s ease 0s;
  transition: border 0.2s ease 0s;
}

<div
    class="emotion-2"
  >
    <select
      class="emotion-0"
    >
      <option
        value="testing"
      >
        Testing
      </option>
    </select>
    <div
      class="emotion-1"
    >
      <svg
        aria-label="arrow double"
        fill="none"
        height="17"
        viewBox="0 0 7 12"
        width="7"
      >
        <path
          d="M0.642491 3.35053L0.292945 3.70804L1.00798 4.40714L1.35752 4.04962L0.642491 3.35053ZM3.75752 1.59491L4.10707 1.23739L3.39204 0.538299L3.04249 0.895815L3.75752 1.59491ZM5.58506 4.04651L5.93149 4.40704L6.65256 3.71417L6.30613 3.35364L5.58506 4.04651ZM3.95354 0.9053L3.6071 0.544767L2.88604 1.23763L3.23247 1.59817L3.95354 0.9053ZM1.35752 7.95041L1.00797 7.59289L0.292938 8.29198L0.642485 8.6495L1.35752 7.95041ZM3.04248 11.1042L3.39203 11.4617L4.10706 10.7626L3.75751 10.4051L3.04248 11.1042ZM6.36054 8.64636L6.70697 8.28583L5.98591 7.59296L5.63947 7.95349L6.36054 8.64636ZM3.28688 10.4018L2.94045 10.7624L3.66152 11.4552L4.00795 11.0947L3.28688 10.4018ZM1.35752 4.04962L3.75752 1.59491L3.04249 0.895815L0.642491 3.35053L1.35752 4.04962ZM6.30613 3.35364L3.95354 0.9053L3.23247 1.59817L5.58506 4.04651L6.30613 3.35364ZM0.642485 8.6495L3.04248 11.1042L3.75751 10.4051L1.35752 7.95041L0.642485 8.6495ZM5.63947 7.95349L3.28688 10.4018L4.00795 11.0947L6.36054 8.64636L5.63947 7.95349Z"
          fill="#000"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
